FrejaEidClient/src/main/java/com/verisec/frejaeid/client/service/AuthenticationService.java

```diff
@@ -101,4 +119,30 @@ private boolean isPollingTimeExpired(long pollingEndTime) {
         return (System.currentTimeMillis() + pollingTimeoutInMilliseconds) < pollingEndTime;
     }
 
+    public static byte[] readAllBytes(InputStream inputStream) throws IOException {
+        final int bufLen = 4 * 0x400; // 4KB
+        byte[] buf = new byte[bufLen];
+        int readLen;
+        IOException exception = null;
+
+        try {
+            try (ByteArrayOutputStream outputStream = new ByteArrayOutputStream()) {
+                while ((readLen = inputStream.read(buf, 0, bufLen)) != -1)
+                    outputStream.write(buf, 0, readLen);
+
+                return outputStream.toByteArray();
+            }
+        } catch (IOException e) {
+            exception = e;
+            throw e;
+        } finally {
+            if (exception == null) inputStream.close();
```

Comment:
Note sure if this if is needed, also this exception logic confuses me a bit.

Reply:
I agree, handling exceptions is a bit confusing, we can go through the code together.

Reply:
Done, removed this part of the code and simplified the function.